storage/collections/ibft_storage.go

```diff
@@ -1,40 +1,145 @@
 package collections
 
 import (
+	"encoding/binary"
+	"encoding/json"
 	"github.com/bloxapp/ssv/ibft/proto"
 	"github.com/bloxapp/ssv/storage"
 	"go.uber.org/zap"
 )
 
 // Iibft is an interface for persisting chain data
 type Iibft interface {
-	SavePrepared(signedMsg *proto.SignedMessage)
-	SaveDecided(signedMsg *proto.SignedMessage)
+	// SaveCurrentInstance saves the state for the current running (not yet decided) instance
+	SaveCurrentInstance(state *proto.State) error
+	// GetCurrentInstance returns the state for the current running (not yet decided) instance
+	GetCurrentInstance(pk []byte) (*proto.State, error)
+	// SaveDecided saves a signed message for an ibft instance with decided justification
+	SaveDecided(signedMsg *proto.SignedMessage) error
+	// GetDecided returns a signed message for an ibft instance which decided by identifier
+	GetDecided(pk []byte, seqNumber uint64) (*proto.SignedMessage, error)
+	// SaveHighestDecidedInstance saves a signed message for an ibft instance which is currently highest
+	SaveHighestDecidedInstance(signedMsg *proto.SignedMessage) error
+	// GetHighestDecidedInstance gets a signed message for an ibft instance which is the highest
+	GetHighestDecidedInstance(pk []byte) (*proto.SignedMessage, error)
 }
 
 // IbftStorage struct
+// instanceType is what separates different iBFT eth2 duty types (attestation, proposal and aggregation)
 type IbftStorage struct {
-	prefix []byte
-	db     storage.Db
-	logger *zap.Logger
+	prefix       []byte
+	instanceType string
+	db           storage.Db
+	logger       *zap.Logger
 }
 
 // NewIbft create new ibft storage
-func NewIbft(db storage.Db, logger *zap.Logger) IbftStorage {
+func NewIbft(db storage.Db, logger *zap.Logger, instanceType string) IbftStorage {
 	ibft := IbftStorage{
-		prefix: []byte("ibft"),
-		db:     db,
-		logger: logger,
+		prefix:       []byte("ibft"),
+		instanceType: instanceType,
```

Comment:
might be worth to change the prefix too or instead of instance type just use differ prefix